dashboard/app/controllers/foorm/forms_controller.rb

```diff
@@ -0,0 +1,82 @@
+module Foorm
+  class FormsController < ApplicationController
+    before_action :require_levelbuilder_mode_or_test_env
+    before_action :authenticate_user!
+    load_and_authorize_resource
+
+    # PUT foorm/form/:id/edit_questions
+    def edit_questions
+      questions_json = get_questions
+      published_state = questions_json['published']
+
+      if published_state.nil_or_empty?
+        questions_json['published'] = @form.published
+      elsif !published_state.nil_or_empty? && !published_state.to_bool && @form.published
```

Comment:
nit: could remove `!published_state.nil_or_empty? &&`, which looks redundant based on the `if` clause?